### PR TITLE
Updated alignment summary stats

### PR DESCRIPTION
**Pull request by pranjan77**
- Removed oracle-java8 dependency. Tests pass. Picard which validates sam file works.
- Updated alignment stats 
- Added properly paired information
- Fixed formatting and unused variables
- Fixed issue with handle delete
- Fixed issues with ws related tests

### CODE REVIEW

File: lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py [hunk 1]
```diff
@@ -242,92 +208,108 @@ def _get_aligner_stats(self, bam_file):
...
-
         elapsed_time = time.time() - start_time
         self.__LOGGER.info('Used: {}'.format(time.strftime("%H:%M:%S", time.gmtime(elapsed_time))))
 
+
         stats_data = {
             "alignment_rate": alignment_rate,
             "mapped_reads": mapped_reads_count,
             "multiple_alignments": multiple_alignments,
-            "properly_paired": properly_paired,
```
Comment (Tianhao-Gu) [inline lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py]: should `properly_paired` be kept in return?

File: lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py [hunk 2]
```diff
@@ -242,92 +208,108 @@ def _get_aligner_stats(self, bam_file):
...
+        secondary_alignment_left_reads_ids = []
+        secondary_alignment_right_reads_ids = []
+        secondary_alignment_single_end_reads_ids = []
+
         mapped_left_reads_ids = []
         mapped_right_reads_ids = []
+        mapped_single_end_reads_ids = []
+
         paired = False
+        pcount = 0
```
Comment (Tianhao-Gu) [inline lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py]: `pcount` is never used.

File: test/ReadsAlignmentUtils_server_test.py [hunk 3]
```diff
@@ -93,8 +93,9 @@ def tearDownClass(cls):
         if hasattr(cls, 'nodes_to_delete'):
             for node in cls.nodes_to_delete:
                 cls.delete_shock_node(node)
+        #TODO:Fix handle delete. Error is handle not owned by user
```
Comment (Tianhao-Gu) [inline test/ReadsAlignmentUtils_server_test.py]: since we fixed it. we can remove `TODO`

File: test/ReadsAlignmentUtils_server_test.py [hunk 4]
```diff
@@ -609,20 +610,23 @@ def test_upload_fail_non_existant_file(self):
             }, self.more_upload_params),
             'File does not exist: foo')
 
+
+#TODO: Following tests fail. They test workspace functionality. Either remove them
+# or fix them so that they don't fail
```
Comment (Tianhao-Gu) [inline test/ReadsAlignmentUtils_server_test.py]: `todo` comments can be removed.

File: lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py (hunk 1 above)
Comment (pranjan77) [reply lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py]: fixed

File: lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py (hunk 2 above)
Comment (pranjan77) [reply lib/ReadsAlignmentUtils/ReadsAlignmentUtilsImpl.py]: fixed

File: test/ReadsAlignmentUtils_server_test.py (hunk 3 above)
Comment (pranjan77) [reply test/ReadsAlignmentUtils_server_test.py]: fixed

File: test/ReadsAlignmentUtils_server_test.py (hunk 4 above)
Comment (pranjan77) [reply test/ReadsAlignmentUtils_server_test.py]: fixed

**Pull request merged**